Ninjathulhu/Ninjathulhu.Game/Component.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace rogue.src.sim
+{
+    class ComponentProperties
+    {
+        public Dictionary<string, object> Values;
+    }
+
+    class Component
+    {
+        protected Entity Entity;
+
+        public bool Enabled;
+
+        ComponentProperties Properties;
+
+        public virtual void Start() {}
+
+        public virtual void Update(float elapsed) {}
+
+        public Component Get<TComponent>()
+            where TComponent : Component
+        {
+            return Entity != null ? Entity.Components.Get<TComponent>() : null;
+        }
+
+        public bool IsSibling(Component other)
+        {
+            return Entity != null && Entity == other.Entity;
+        }
+
+        public virtual void HandleAttachToEntity(Entity entity)
```

Comment:
I'd probably call this `AttachTo`; if we use it like `component.AttachTo(entity)` it reads pretty well.